### PR TITLE
Callback validator consumes string class name via ::class resolution

### PR DESCRIPTION
**Pull request by vaclavvanik**
Added ability to inject string class name into callback validator constructor.

### CODE REVIEW

File: doc/book/validators/callback.md [hunk 1]
```diff
@@ -123,6 +123,26 @@ if ($valid->isValid($input)) {
...
+class MyClass
+{
+    public function __invoke($value)
+    {
+        // some validation
+        return true;
+    }
+}
+
+$valid = new Zend\Validator\Callback(MyClass::class);
```
Comment (Ocramius) [inline doc/book/validators/callback.md]: This needs the method to be static...

File: src/Callback.php [hunk 2]
```diff
@@ -74,6 +74,10 @@ public function getCallback()
      */
     public function setCallback($callback)
     {
+        if (is_string($callback) && class_exists($callback)) {
+            $callback = new $callback();
```
Comment (Ocramius) [inline src/Callback.php]: This sort of magic makes it unclear when DI can be used or not

File: doc/book/validators/callback.md (hunk 1 above)
Comment (Ocramius) [inline doc/book/validators/callback.md]: Nvm, was confused, then read the code for it. This breaks the principle of least possible astonishment for me.

**Comment by vaclavvanik** [comment]
Im not happy that `setCallback` brings more magic. However anonymous functions cannot be serialized (and pulled from merged application config), this PR brings possibility to bypass serialization restriction. I hope that reason for this PR is more obvious.

**Comment by vaclavvanik** [comment]
ping @weierophinney

**Comment by vaclavvanik** [comment]
Pinging @Ocramius and @weierophinney for review.

**Comment by Ocramius** [comment]
As mentioned above, I don't like this auto-instantiating approach. Most callbacks require some sort of DB connection, lookup system, etc, so they will unlikely fit this particular validator.

Generally 👎 from me, sorry.

**Comment by Xerkus** [comment]
Anonymous functions can be injected via `Module::getValidatorConfig()` in zend-modulemanager based projects,  besides your specific use case would be better with static `['Class', 'method']` callable anyway.

:-1: from me as well, sorry

**Pull request closed without merging**